### ISSUE TITLE
s3 etcd backups taking longer than 5 minutes cause backup loop

### ISSUE THREAD

**Issue by nickgerace** [rancher/rancher#34890]
Backups that take longer than 5 minutes will cause Rancher to start a new backup and delete the one that is currently running. Since they all take longer than 5 minutes it will continue this cycle indefinitely.

**Rancher Server Setup**
- Rancher version: 2.6.3

**Repro steps:**
* Have RKE1 provisioned downstream cluster
* Fill etcd with lots of data (e.g. 1 GB+) - enough so it takes more than 5 minutes to complete on all of the etcd nodes.
* Configure recurring snapshots for this downstream cluster, including setting `rancher_kubernetes_engine_config.services.etcd.backup_config.timeout` to high enough value for backups to actually complete on etcd side (e.g. 900 for 15 minutes).
* Wait for a backup to start (depending on interval configured) and wait 5 minutes.

**Actual behavior:**
Since the backup doesn't complete before the sync interval, a new backup will be started. This will repeat indefinitely until a backup is completed in under 5 minutes.

**Expected behavior:**
Backups to complete.

Original description (no longer relevant as fix approach changed per https://github.com/rancher/rancher/issues/34890#issuecomment-943872638):
> The `clusterBackupCheckInterval` is hardcoded to 5 minutes and should be user-changeable via the following
> - Helm (likely)
> - ENV VAR (could be good idea)
> - ConfigMap (eh)
> - etc.
> 
> Moreover, we should considered adding a "maximum retry count" so that the check interval does not infinitely continue.

JIRA ID: SURE-3305

**Comment by nickgerace** [comment]
The "part two" to this issue exists and is listed below, but requires a design discussion and investigation. It is _not_ required for solving this issue.

"Part two" issue: https://github.com/rancher/rancher/issues/34892

**Comment by bentastic27** [comment]
Less intrusive PR that should be far more reliable. Basically a check for running backups instead of bulldozing them and starting over.

https://github.com/rancher/rancher/pull/35132

It doesn't make the clusterBackupCheckInterval malleable, but IMO it makes it far less an issue, perhaps even unnecessary. It appears that changing the check interval it could interfere with the backup interval setting. Let's say that a user sets a backup to run every hour, but they set the check interval to 2 hours. It looks as though that the backup would only run 2 hours if that was the case, so the check interval can be considered resolution of sorts in this case:

By default a backup will run every hour with 5 minutes of slop.

**Comment by snasovich** [comment]
@SheilaghM , moved it to `team/area2` as it's etcd snapshot/restore and not Rancher's backup/restore operator.

**Comment by snasovich** [comment]
Assigning myself and Jake for visibility to get the PR merged and update issue description if we're going with this current approach by @bentastic27 .

**Comment by snasovich** [comment]
The issue is ready for testing on v2.6-head (3dbea4c3e70df59f50ce161b70d0b5e1da9319a4)

### Root cause
If a user sets the backup timeout to 10 minutes due to a slow S3-alike or similar, the backup will still be running during the next tick of the check interval (every 5 minutes). The cluster backup sync function will see that the last completed backup was longer than the backup interval (ignoring the one that's currently running) so it'll start a new backup and bulldoze the running one.
 
### What was fixed, or what changes have occurred
Check that if backup's creation time is less than the backup config's timeout and "completed" is unknown, then skip until the next iteration. 

### Areas or cases that should be tested
Recurring etcd backups for RKE1 clusters
 
### What areas could experience regressions?
Recurring etcd backups
 
### Are the repro steps accurate/minimal?
Yes (however note the exact amount of data necessary in etcd is unclear and depends on many factors - just need the backup to take over 5 minutes - exercise trial and error to find the amount).

**Comment by snasovich** [comment]
@Auston-Ivison-Suse @slickwarren , not sure who has the capacity between you two to validate this. Please reproduce this issue on v2.6.3 and ensure it's not reproducible on v2.6-head.

**Comment by rancherbot** [comment]
@snasovich, Not creating `backport` issue for issue `#34890` in repository `rancher` because milestone `2.5.12` does not exist or is not an open milestone

**Comment by rancherbot** [comment]
@snasovich, Not creating `backport` issue for issue `#34890` in repository `rancher` because milestone `2.5.12` does not exist or is not an open milestone

**Comment by slickwarren** [comment]
after some more testing, I was still unable to reproduce the original issue.  We currently don't have an exact way to saturate etcd to the level that the customer reported, so we tried the following workaround with no luck reproducing:

test 1 -- throttle upload to s3 by adding a proxy in front of the s3 server -- tested on on 2.5-head (923f59cb62731d275d050802ab403153eb37d8ac) and 2.6.3
* throttle connection to your s3 store to 20kbps
* set timeout to > 5 minutes (I used 3600 seconds)
* ensure that fresh install backup completes successfully
* saturate etcd with some secrets (~500)
* take another snapshot 
* RESULT: snapshot uploads to s3 store, but etcd snapshot remains in `activating` state through rancher 
  * persists through refreshes
  * persists after new timeout limit is reached
  * only way to get it out of this state that I found was to increase the connection throttle to a greater value
  * the following error is observed in the rancher logs: 
```
Waiting for [etcd-snapshot-once] container to exit on host [<rancher-hostname>]
Container [etcd-snapshot-once] is still running on host [<rancher-hostname>]: stderr: [time="<current_time>" level=info msg="failed to upload etcd snapshot file: 504 Gateway Time-out, retried 1 times"
```
which repeats indefinitely, from 0 to 1 to 2 `retried <> times`

Perhaps this is closer to what the customer encountered, as even after 5 minutes I am able to see the backup file is successfully uploaded in the s3 store on 2.6.3.

**Comment by slickwarren** [comment]
blocked (backport has an issue, waiting on it to retest) https://github.com/rancher/rancher/issues/36038

**Comment by slickwarren** [comment]
tested on v2.6-head (c682e30):
* deploy downstream cluster configured to backup to proxied minio (set backup timeout to 1 hour)
  * initial backup should complete in less than 5 mintues -- pass
  * saturate etcd, making backup take longer than 5 minutes, and perform manual backup -- pass
  * wait for automatic backup to begin and complete (automatic backup should take longer than 5 minutes) -- pass
  * restore from initial backup -- pass
  * restore from saturated backup -- pass

related issue: https://github.com/rancher/rancher/issues/36209
